docs/samples/line/labelVisibility.md

```diff
@@ -23,29 +23,57 @@ const data = {
 };
 // </block:setup>
 
-// <block:annotation:1>
-const annotation = {
+// <block:annotation1:1>
+const annotation1 = {
   type: 'line',
   borderColor: 'lightGreen',
-  borderDashOffset: 0,
   borderWidth: 10,
-  drawTime: 'beforeDatasetsDraw',
   label: {
     enabled: false,
     backgroundColor: 'green',
-    borderWidth: 0,
     drawTime: 'afterDatasetsDraw',
-    color: 'white',
-    content: (ctx) => ['Average of dataset', 'is: ' + average(ctx).toFixed(3)],
-    textAlign: 'center'
+    content: (ctx) => ['Average of dataset', 'is: ' + average(ctx).toFixed(3)]
   },
   scaleID: 'y',
   value: (ctx) => average(ctx),
+  // For simple property changes, you can directly modify the annotation
+  // element's properties then call chart.draw().  This is faster.
+  enter({chart, element}, event) {
+    element.options.label.enabled = true;
+    chart.draw();
+  },
+  leave({chart, element}, event) {
+    element.options.label.enabled = false;
+    chart.draw();
+  }
+};
+// </block:annotation>
+
+// <block:annotation2:2>
+const annotation2 = {
+  type: 'line',
+  borderColor: 'lightBlue',
+  borderWidth: 10,
+  label: {
+    enabled: (ctx) => ctx.hovered,
+    backgroundColor: 'blue',
+    drawTime: 'afterDatasetsDraw',
+    content: (ctx) => ['Min of dataset', 'is: ' + min(ctx).toFixed(3)],
+    position: (ctx) => ctx.hoverPosition
+  },
+  scaleID: 'y',
+  value: (ctx) => min(ctx),
+  // For more complex dynamic properties, you can store values on the persistent
+  // context object then retrieve them via scriptable properties.  You'll have
+  // to call chart.update() to reprocess the chart.
   enter(ctx, event) {
-    toggleLabel(ctx, event);
+    ctx.hovered = true;
+    ctx.hoverPosition = (event.x / ctx.chart.chartArea.width * 100) + '%';
+    ctx.chart.update();
   },
   leave(ctx, event) {
-    toggleLabel(ctx, event);
+    ctx.hovered = false;
+    ctx.chart.update();
   }
 };
 // </block:annotation>
```

Comment:
I think it should be changed the annotation tag to `// </block:annotation2>` otherwise it's shown in the example source  as comment.

Reply:
Fixed. Thanks.